addons/l10n_in/static/src/js/hsn_autocomplete.js

```diff
@@ -0,0 +1,100 @@
+/** @odoo-module **/
+
+import { AutoComplete } from "@web/core/autocomplete/autocomplete";
+import { registry } from "@web/core/registry";
+import { _t } from "@web/core/l10n/translation";
+import { CharField, charField } from "@web/views/fields/char/char_field";
+
+const l10n_in_hsn_service_url = "https://services.gst.gov.in/commonservices/hsn/search/qsearch";
+
+export class L10nInHsnAutoComplete extends CharField {
+    static template = "hsn_autocomplete.L10nInHsnAutoComplete";
+    static components = {
+        ...CharField.components,
+        AutoComplete,
+    };
+    static props = {
+        ...CharField.props,
+        l10n_in_hsn_description: { type: String, optional: true },
+    };
+
+    setup() {
+        super.setup();
+    }
+
+    async getHsnSuggestions(value) {
+        const isDigit = [...value].every((v) => !isNaN(v) && v !== " ");
+
+        const allUrlAndParams = [
+            { url: "byCode", category: "null" },
+            { url: "byDesc", category: "P" },
+            { url: "byDesc", category: "S" },
+        ];
+
+        const filteredUrlsAndParams = isDigit ? [allUrlAndParams[0]] : allUrlAndParams.slice(1, 3);
+
+        for (const { url, category } of filteredUrlsAndParams) {
+            try {
+                const queryParams = new URLSearchParams({
+                    inputText: value,
+                    selectedType: url,
+                    category: category,
+                });
+                const response = await fetch(`${l10n_in_hsn_service_url}?${queryParams}`);
```

Comment:
From my test it seems like products and services, so I guess we would need to do both calls ? For efficiency Could we use `Promise.all` instead of waiting for one then waiting for the second.

Reply:
Any reason why it's either byCode or byDesc and not both? they seem to return different results
I tried:
https://services.gst.gov.in/commonservices/hsn/search/qsearch?selectedType=ByDesc&inputText=7112&category=P
https://services.gst.gov.in/commonservices/hsn/search/qsearch?selectedType=ByCode&inputText=7112&category=null

Reply:
done

Reply:
> I don't think the try/catch is necessary - please check which is better between `promise.all` and `promise.allSettled`

What if the URL change?, `fetch` throws an error